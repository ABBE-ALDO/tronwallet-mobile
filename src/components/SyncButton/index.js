import React, { PureComponent } from 'react'
import { Animated, TouchableOpacity, Easing, StyleSheet } from 'react-native'

import { Spacing } from '../DesignSystem'

class SyncButton extends PureComponent {
  static defaultProps = {
    onPress: () => null
  }

  constructor () {
    super()
    this.spinValue = new Animated.Value(0)
  }

  componentDidMount () {
    this._spin()
  }

  componentDidUpdate (prevProps, prevState, snapshot) {
    const { loading } = this.props
    if (prevProps.loading !== loading) {
      if (loading) this._spin()
    }
  }

  _spin = () => {
    const { loading } = this.props
    if (!loading) return
    this.spinValue.setValue(0)
    Animated.timing(
      this.spinValue,
      {
        toValue: 1,
        duration: 900,
        easing: Easing.linear,
        useNativeDriver: true
      }
    ).start(() => this._spin())
  }

  render () {
    const { onPress, loading } = this.props
    const spin = this.spinValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '360deg']
    })

    return (
      <TouchableOpacity disabled={loading} style={styles.touchable} onPress={onPress}>
        <Animated.Image
          style={[styles.image, {
            opacity: loading ? 0.4 : 1,
            transform: [{ rotate: spin }]
          }]}
          source={require('../../assets/refresh.png')}
        />
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  touchable: {
    padding: Spacing.medium
  },
  image: {
    width: 17,
    height: 17
  }
})

export default SyncButton
