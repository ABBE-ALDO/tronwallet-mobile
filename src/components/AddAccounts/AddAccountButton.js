import React from 'react'
import { TouchableOpacity } from 'react-native'
import Feather from 'react-native-vector-icons/Feather'
import { func, bool, string } from 'prop-types'

import SyncButton from '../SyncButton'

const AddAccountButton = ({ onPress, loading, secretMode }) => {
  if (secretMode === 'privatekey') {
    return null
  }

  return (
    <TouchableOpacity onPress={onPress} disabled={loading} >
      {loading
        ? (<SyncButton loading />)
        : (<Feather name='plus' color={'white'} size={28} />)}
    </TouchableOpacity>
  )
}

AddAccountButton.propTypes = {
  onPress: func,
  loading: bool,
  secretMode: string.isRequired
}

AddAccountButton.defaultProps = {
  onPress: () => {},
  loading: false
}

export default AddAccountButton
