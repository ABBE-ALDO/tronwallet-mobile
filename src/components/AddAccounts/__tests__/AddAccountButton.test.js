import React from 'react'
import { TouchableOpacity } from 'react-native'
import Feather from 'react-native-vector-icons/Feather'
import { shallow } from 'enzyme'

import AddAccountButton from '../AddAccountButton'
import SyncButton from '../../SyncButton'

jest.mock('../../../utils/i18n')

describe('Account button component', () => {
  const setup = propOverrides => {
    const props = Object.assign({
    }, propOverrides)

    const wrapper = shallow(
      <AddAccountButton {...props} />
    )

    return {
      props,
      wrapper
    }
  }

  test('Should match when render component', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('Should show button when secret mode is default value', () => {
    const { wrapper } = setup()

    expect(wrapper.find(TouchableOpacity).exists()).toBe(true)
  })

  test('Should show button when secret mode is mnemonic', () => {
    const { wrapper } = setup({ secretMode: 'mnemonic' })

    expect(wrapper.find(TouchableOpacity).exists()).toBe(true)
  })

  test('Should hide button when secret mode is privatekey', () => {
    const { wrapper } = setup({ secretMode: 'privatekey' })

    expect(wrapper.find(TouchableOpacity).exists()).toBe(false)
  })

  test('Should show plus icon when loading is default value', () => {
    const { wrapper } = setup()

    expect(wrapper.find(Feather).exists()).toBe(true)
  })

  test('Should show plus icon when loading is false', () => {
    const { wrapper } = setup({ loading: false })

    expect(wrapper.find(Feather).exists()).toBe(true)
  })

  test('Should show sync icon when loading is true', () => {
    const { wrapper } = setup({ loading: true })

    expect(wrapper.find(SyncButton).exists()).toBe(true)
  })
})
