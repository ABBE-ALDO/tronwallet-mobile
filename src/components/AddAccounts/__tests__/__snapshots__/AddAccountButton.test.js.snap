// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account button component Should match when render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddAccountButton
    loading={false}
    onPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Icon
        allowFontScaling={false}
        color="white"
        name="plus"
        size={28}
      />,
      "disabled": false,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": false,
        "color": "white",
        "name": "plus",
        "size": 28,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Icon
          allowFontScaling={false}
          color="white"
          name="plus"
          size={28}
        />,
        "disabled": false,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": false,
          "color": "white",
          "name": "plus",
          "size": 28,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
